code/game/machinery/medal_printer.dm

```diff
@@ -42,3 +43,12 @@
 	)
 	RefreshParts()
 
+/obj/machinery/r_n_d/attackby(var/obj/item/O as obj, var/mob/user as mob)
+	. = ..()
+
+	if(istype(O, /obj/item/clothing/accessory/medal))
+		O.name = sanitize((input(usr, "What would you like to label [O]?", "Medal Labelling", null)  as text), 1, MAX_NAME_LEN)
+		if((loc == usr && !usr.isUnconscious()))
```

Comment:
This will actually reset the name of the medal `unless` the user is unconscious and the medal has been thrown away.
Did you forget a`!` before that block ?

Reply:
fixed